src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.RawSocket.cs

```diff
@@ -199,9 +199,7 @@ private async Task<PingReply> SendIcmpEchoRequestOverRawSocketAsync(IPAddress ad
             using (Socket socket = GetRawSocket(socketConfig))
             {
                 int ipHeaderLength = socketConfig.IsIpv4 ? MinIpHeaderLengthInBytes : 0;
-                CancellationTokenSource timeoutTokenSource = new CancellationTokenSource();
-
-                timeoutTokenSource.CancelAfter(timeout);
+                CancellationTokenSource timeoutTokenSource = new CancellationTokenSource(timeout);
```

Comment:
Why not just use `using` here?

Reply:
Because we already have a try/catch we can just add a finally to... might as well do that rather than wrapping another try/finally around the existing try/catch.

Reply:
Thought you would say that :-)

Counterargument:
- There's another `using` right above it. Doesn't the compiler coalesce the generated try/finally for successive using blocks?
- Some analyzers from FxCOP time used to flag missing `using` on local objects that don't escape the function. I am not sure whether they can recognise the manual try/finally/dispose pattern.

Reply:
> Doesn't the compiler coalesce the generated try/finally for successive using blocks?

No, it can't, as Dispose could throw on one thing and it should still invoke Dispose on the other.

> Some analyzers from FxCOP time used to flag missing using on local objects that don't escape the function. I am not sure whether they can recognise the manual try/finally/dispose pattern.

They'd be pretty broken and unusable if they didn't :)

Reply:
I guess that in the end it doesn't matter much. I find the `using` keyword easier to read and it's less C# code (the non-block C# 8 syntax) but I guess you are right about the IL size. Thanks for fixing it!